src/test/java/com/epam/edumanagementsystem/admin/rest/repository/SuperAdminRepositoryTest.java

```diff
@@ -0,0 +1,31 @@
+package com.epam.edumanagementsystem.admin.rest.repository;
+
+import com.epam.edumanagementsystem.admin.model.entity.SuperAdmin;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.TestPropertySource;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@TestPropertySource(properties = {"compile('org.flywaydb:flyway-core')"})
+class SuperAdminRepositoryTest {
```

Comment:
Not sure about some annotations, please explain their usage.

Reply:
@AutoConfigureTestDatabase is used to avoid from configuring and using an embedded H2 db by @DataJpaTest annotation. If not there would be sth like "No DataSource detected exception" thrown, due to the fact that there is no H2 db found in classpath. So to avoid it, we should enable the autoconfiguration for configuring an embedded db for test. With current approach the test uses the real db but with Hibernate's ddl : create-drop approach and also rolls back the transaction. @TestPropertySource annotation with properties about flyway is unnecessary.